### PR TITLE
refactor: delete Azure Stack's KubernetesClient implementation

### PR DESCRIPTION
**Pull request by jadarsie**
**Reason for Change**:

Files `pkg/armhelpers/kubeclient.go` and `pkg/armhelpers/azurestack/kubeclient.go` look almost identical. I do not see a good reason to keep two different implementations of interface `armhelpers.KubernetesClient`.

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [x] No
- [ ] Yes

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
- [ ] includes documentation
- [ ] adds unit tests
- [x] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/armhelpers/azureclient.go [hunk 1]
```diff
@@ -80,6 +80,11 @@ type AzureClient struct {
 	servicePrincipalsClient graphrbac.ServicePrincipalsClient
 }
 
+// GetKubernetesClient returns a KubernetesClient hooked up to the api server at the apiserverURL.
```
Comment (jackfrancis) [inline pkg/armhelpers/azureclient.go:83]: Thanks for moving this `AzureClient` pointer method here alongside its other methods. (Unfortunately this should never have been implemented as a method, as it neither operates upon nor modifies the instance object. :/ (But we don't want to worry about that now)

File: pkg/armhelpers/azurestack/azureclient.go [hunk 2]
```diff
@@ -14,6 +14,7 @@ import (
 	"strings"
 	"time"
 
+	"github.com/Azure/aks-engine/pkg/armhelpers"
```
Comment (jackfrancis) [inline pkg/armhelpers/azurestack/azureclient.go:17]: The only thing I'd say here is that this "child" package depending upon the parent can lead to circular dependencies in the future. Ideally we'd put `NewKubernetesClientSetClient` somewhere that both `pkg/armhelpers` and `pkg/armhelpers/azurestack` could import in common. Not sure if there's a candidate package for that?

Comment (jadarsie) [reply pkg/armhelpers/azurestack/azureclient.go:17]: yeah, I thought about that, I was not sure if moving more stuff around was desired, I will look for a new home for this type.

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Pull request merged**